eslint-plugin-expensify/prefer-shouldUseNarrowLayout-instead-of-isSmallScreenWidth.js

```diff
@@ -0,0 +1,99 @@
+const {AST_NODE_TYPES} = require("@typescript-eslint/utils");
+const _ = require("underscore");
+const CONST = require("./CONST");
+
+module.exports = {
+  meta: {
+    type: "problem",
+    docs: {
+      description:
+        "Warn against using isSmallScreenWidth from useResponsiveLayout and suggest using shouldUseNarrowLayout instead.",
+    },
+    schema: [],
+  },
+  create(context) {
+    return {
+      VariableDeclarator(node) {
+        if (
+          !node.init ||
+          !node.init.callee ||
+          node.init.callee.name !== "useResponsiveLayout"
+        ) {
+          return;
+        }
+
+        // Check for 'const {isSmallScreenWidth, ...} = useResponsiveLayout();' pattern
+        if (node.id.type === AST_NODE_TYPES.ObjectPattern) {
+          node.id.properties.forEach((property) => {
+            if (!property.key || property.key.name !== "isSmallScreenWidth") {
+              return;
+            }
+            context.report({
+              node: property,
+              message:
+                CONST.MESSAGE
+                  .PREFER_SHOULD_USE_NARROW_LAYOUT_INSTEAD_OF_IS_SMALL_SCREEN_WIDTH,
+            });
+          });
+        }
+
+        // Check for 'const var = useResponsiveLayout();' and use of this var
+        const variableName = node.id.name;
+        const variableUsages = _.filter(
+          context.getScope().references,
```

Comment:
```suggestion
          sourceCode.getScope().references,
```

can we do this? there is a warning

![Screenshot 2024-10-03 at 11 18 25 in the morning](https://github.com/user-attachments/assets/24b2e6c4-5365-4a52-80ad-3e4a0ac79399)